packages/vite-plugin-svelte/src/index.ts

```diff
@@ -70,7 +70,13 @@ export function svelte(inlineOptions?: Partial<Options>): Plugin {
 		},
 
 		async buildStart() {
-			await handleOptimizeDeps(options, viteConfig);
+			if (!options.experimental.prebundleSvelteLibraries) return;
+			const isSvelteMetadataChanged = await saveSvelteMetadata(viteConfig.cacheDir, options);
+			if (isSvelteMetadataChanged) {
+				// Force Vite to optimize again. Although we mutate the config here, it works because
+				// Vite's optimizer runs after `buildStart()`.
+				viteConfig.server.force = true;
```

Comment:
Is this safe? What happens when the server is restarted afterwards

Reply:
Hmm yeah. Looks like if the server restart later on, it would always issue force optimization. Maybe reverting this on `buildEnd()` would fix this. I'll test this later on.

Reply:
I just tested this and the `buildEnd()` thing isn't needed. It looks like Vite re-constructs the config on restart so this change won't be carried on after restarts. With that said, I think this trick should still be safe and there isn't any other way to achieve the behaviour, since Vite 2.9 prebundles dep in a non-blocking way now.

Reply:
Feels hacky still. Is it possible to use server.restart or add api for this to vite?

Reply:
`server.restart` is tricky since we don't have the new resolved config before determining to pass `server.restart(true)` or not. And that would handle the case while the dev server is running and restart. If the user changed the config and then start the dev server, we'd have to checked that too externally.

Perhaps an API vite can provide is `server.forceOptimize` but I'll have to look into that. The trick here is actually borrowed from [here](https://github.com/vitejs/vite/blob/a74bd7ba9947be193bccf636b8918bd1f59e89ae/packages/vite/src/node/server/index.ts#L736-L742).

Reply:
hmm, it feels like this warrants/needs new api in vite. plugins contributing to prebundling/optimization needing to hack things doesn't feel like it's a long term solution. with vite3 coming up maybe it would be a good time to do something about it?

while it may work now it'll break the second someone refactors the logic around vite server.force. We might learn about it in vite-ecosystem-ci before it actually burns users but this creates a coupling i'm not comfortable with :/

Reply:
Yeah. I'll take a look at supporting an official API at the meantime. I have some ideas about it which might work, and having a safe API for plugins to use would be good for vue/marko adoption too.

I'm not sure how long that'll take though, once I got something up I'll check with patak to see if we can ship it soon.

Perhaps we can still merge this given it's behind an experimental flag. Probably won't turn it on for SvelteKit yet, but with the edge cases resolved we may be able to get more feedback. I'll also be keeping an eye in Vite if anyone tries to hurt my hack 😄

Reply:
as long as it stays behind an experimental flag i'm fine with it (: +1: